library/src/main/java/io/github/fanky10/sociallogin/module/fragments/BaseGooglePlusLoginFragment.java

```diff
@@ -32,14 +73,92 @@ public void doGoogleLogin() {
     }
 
     private void doGoogleLoginProcess() {
-        mGooglePlusProvider = new GooglePlusProvider(getActivity());
-        mGooglePlusProvider.setCallback(this);
+        if (mGoogleApiClient != null) {
+            if (!mGoogleApiClient.isConnected()) {
+                mGoogleApiClient.connect();
+            }
+        }
+    }
+
+    @Override
+    public void onConnected(Bundle bundle) {
+        Plus.PeopleApi.load(mGoogleApiClient, "signed_in_user_account_id").setResultCallback(this);
+    }
+
+    @Override
+    public void onConnectionSuspended(int i) {
+        onSocialProviderConnectionCanceled();
+    }
+
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+        if (connectionResult.hasResolution()) {
+            try {
+                connectionResult.startResolutionForResult(getActivity(), REQUEST_RESOLVE_ERROR);
+            } catch (IntentSender.SendIntentException e) {
+                e.printStackTrace();
```

Comment:
I think here we should fire on social provider connection failure, I am correct? may be it is still recoverable.